src/libraries/System.Runtime.Serialization.BinaryFormat/src/PayloadReader.cs

```diff
@@ -0,0 +1,285 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace System.Runtime.Serialization.BinaryFormat;
+
+#if SYSTEM_RUNTIME_SERIALIZATION_BINARYFORMAT
+public
+#else
+internal
+#endif
+static class PayloadReader
+{
+    private static UTF8Encoding ThrowOnInvalidUtf8Encoding { get; } = new(false, throwOnInvalidBytes: true);
+
+    /// <summary>
+    /// Checks if given buffer contains only Binary Formatter payload.
+    /// </summary>
+    /// <param name="bytes">The buffer to inspect</param>
+    /// <returns>True if the first and last bytes indicate Binary Format, otherwise false.</returns>
+    public static bool ContainsBinaryFormatterPayload(ReadOnlySpan<byte> bytes)
+        => bytes.Length >= SerializedStreamHeaderRecord.Size + 2
+            && bytes[0] == (byte)RecordType.SerializedStreamHeader
+            && bytes[bytes.Length - 1] == (byte)RecordType.MessageEnd;
+
+    /// <summary>
+    /// Checks if given stream contains only Binary Formatter payload.
+    /// </summary>
+    /// <param name="stream">The readable and seekable stream to inspect.</param>
+    /// <returns>True if the first and last bytes indicate Binary Format, otherwise false.</returns>
+    /// <exception cref="ArgumentNullException">The stream is null.</exception>
+    /// <exception cref="NotSupportedException">The stream does not support reading or seeking.</exception>
+    /// <exception cref="ObjectDisposedException">The stream was closed.</exception>
+    /// <remarks><para>It does not modify the position of the stream.</para></remarks>
+    public static bool ContainsBinaryFormatterPayload(Stream stream)
+    {
+#if NETCOREAPP
+        ArgumentNullException.ThrowIfNull(stream);
+#else
+        if (stream is null) throw new ArgumentNullException(nameof(stream));
```

Comment:
This line is prohibited by the style guide.  https://github.com/dotnet/runtime/blob/7fdc74eb68a75c98c8797fc9c58878c05e128316/docs/coding-guidelines/coding-style.md, rule 18, bullet 1.

```suggestion
        if (stream is null)
            throw new ArgumentNullException(nameof(stream));
```